PyPDF2/_utils.py

```diff
@@ -241,17 +241,13 @@ def paeth_predictor(left: int, up: int, up_left: int) -> int:
         return up_left
 
 
+def deprecate(msg: str, stacklevel: int = 3) -> None:
+    warnings.warn(msg, PendingDeprecationWarning, stacklevel=stacklevel)
```

Comment:
One note is that I believe that this should be changed to a `DeprecationWarning` as they've been deprecated and will be removed in next major version, where `PendingDeprecationWarning` should be used for functions that will be deprecated in the future, but are not yet.

Reply:
I'm fine with changing it.

As we are now on it, maybe we should specify the process. I would suggest the following if we want to remove a class / function / method. If the version before the change is `x.y.z`, then:

1.  `x.y.(z+1)`: Add a `PendigDeprecationWarning`. If there is a replacement, the replacement is also introduced and the warning informs about this change. The docs let users know about the deprecation + the new function. The CHANGELOG informs about it.
2. `(x+1).0.0`: The `PendingDeprecationWarning` is changed to a `DeprecationWarning`. The CHANGELOG informs about it.
3. `(x+2).0.0`: The code is actually removed. The CHANGELOG informs about it.

Reply:
I've added those thoughts to a PR: https://github.com/py-pdf/PyPDF2/pull/930